src/mpid/ch4/src/ch4i_workq.h

```diff
@@ -13,5 +13,133 @@
 #define CH4I_WORKQ_H_INCLUDED
 
 #include "ch4i_workq_types.h"
+#include <utlist.h>
+
+#if defined(MPIDI_CH4_USE_WORK_QUEUES)
+struct MPIDI_workq_elemt MPIDI_workq_elemt_direct[MPIDI_WORKQ_ELEMT_PREALLOC];
+extern MPIR_Object_alloc_t MPIDI_workq_elemt_mem;
+
+MPL_STATIC_INLINE_PREFIX struct MPIDI_workq_elemt *MPIDI_workq_elemt_create(void)
+{
+    return MPIR_Handle_obj_alloc(&MPIDI_workq_elemt_mem);
+}
+
+MPL_STATIC_INLINE_PREFIX void MPIDI_workq_elemt_free(struct MPIDI_workq_elemt *elemt)
+{
+    MPIR_Handle_obj_free(&MPIDI_workq_elemt_mem, elemt);
+}
+
+MPL_STATIC_INLINE_PREFIX void MPIDI_workq_pt2pt_enqueue(MPIDI_workq_t * workq,
+                                                        MPIDI_workq_op_t op,
+                                                        const void *send_buf,
+                                                        void *recv_buf,
+                                                        MPI_Aint count,
+                                                        MPI_Datatype datatype,
+                                                        int rank,
+                                                        int tag,
+                                                        MPIR_Comm * comm_ptr,
+                                                        int context_offset,
+                                                        MPIDI_av_entry_t * addr,
+                                                        MPI_Status * status,
+                                                        MPIR_Request * request,
+                                                        int *flag,
+                                                        MPIR_Request ** message,
+                                                        OPA_int_t * processed)
+{
+    MPIDI_workq_elemt_t *pt2pt_elemt;
+
+    MPIR_Assert(request != NULL);
+
+    MPIR_Request_add_ref(request);
+    pt2pt_elemt = &request->dev.ch4.command;
+    pt2pt_elemt->op = op;
+    pt2pt_elemt->processed = processed;
+    pt2pt_elemt->pt2pt.send_buf = send_buf;
+    pt2pt_elemt->pt2pt.recv_buf = recv_buf;
+    pt2pt_elemt->pt2pt.count = count;
+    pt2pt_elemt->pt2pt.datatype = datatype;
+    pt2pt_elemt->pt2pt.rank = rank;
+    pt2pt_elemt->pt2pt.tag = tag;
+    pt2pt_elemt->pt2pt.comm_ptr = comm_ptr;
+    pt2pt_elemt->pt2pt.context_offset = context_offset;
+    pt2pt_elemt->pt2pt.addr = addr;
+    pt2pt_elemt->pt2pt.status = status;
+    pt2pt_elemt->pt2pt.request = request;
+    pt2pt_elemt->pt2pt.flag = flag;
+    pt2pt_elemt->pt2pt.message = message;
+
+    MPIDI_workq_enqueue(workq, pt2pt_elemt);
```

Comment:
I did not see the `MPIDI_workq_enqueue` function in this PR. Missed in `git add`?

Reply:
There is, although it is a stub at this moment. See the patch `ch4: Add work queue type definitions`. It'll be replaced by one from izem.

Reply:
OK.